arborist/server.go

```diff
@@ -334,20 +334,31 @@ func (server *Server) handleAuthProxy(w http.ResponseWriter, r *http.Request) {
 	authRequest.stmts = server.stmts
 	w.Header().Set("REMOTE_USER", authRequest.Username)
 
-	rv, err := authorizeUser(authRequest)
-	if err != nil {
-		msg := fmt.Sprintf("could not authorize user: %s", err.Error())
-		server.logger.Info("tried to handle auth request but input was invalid: %s", msg)
-		response := newErrorResponse(msg, 400, nil)
-		_ = response.write(w, r)
+	if (authRequest.Username == "") && (authRequest.ClientID == "") {
+		msg := "unauthorized: did not provide a username and/or client ID in request"
+		_ = newErrorResponse(msg, 403, nil).write(w, r)
 		return
 	}
-	if rv.Auth {
-		server.logger.Debug("user is authorized")
-	} else {
-		server.logger.Debug("user is unauthorized")
+
+	rv := &AuthResponse{}
+	rv.Auth = true
+	var err error = nil
+	if authRequest.Username != "" {
```

Comment:
quick question about the logic here, do we authorize based on either username or client id? In that case can we rewrite the code to be something like the below, it'll be a bit cleaner

```
rv := &AuthResponse{}
rv.Auth = true
if authRequest.Username != "" {
  rv, err = authorizeUser()
}
else {
  rv, err = authorizeClient()
}

if err != nil 
if err != nil {
			msg := fmt.Sprintf("could not authorize user: %s", err.Error())
			server.logger.Info("tried to handle auth request but input was invalid: %s", msg)
			response := newErrorResponse(msg, 400, nil)
			_ = response.write(w, r)
			return
		}
		if rv.Auth {
			server.logger.Debug("user is authorized")
		} else {
			server.logger.Debug("user is unauthorized")
		}

```

Reply:
no, we authorize based on both username and client_id, unless there is only one of the 2, in which case we authorize baed on only the one that's provided